src/signature.rs

```diff
@@ -190,45 +183,23 @@ impl Signature {
             Err(x) => return Err(x),
         }
 
-        Ok(Signature {
+        Ok(InternalSignature {
             R: CompressedEdwardsY(lower),
             s: s,
         })
     }
 }
 
-#[cfg(feature = "serde")]
-impl Serialize for Signature {
```

Comment:
Note: the `ed25519` crate has its own `serde` impls, so these are no longer necessary